### PR TITLE
[INTERNAL]  ignore package-lock.json file

### PR DESCRIPTION
**Pull request by cstrnt**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Additional info:**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
     "target": "ESNext",
     "module": "ESNext",
     "lib": ["ESNext"],
-    "noImplicitAny": false
+    "noImplicitAny": false,
+    "allowSyntheticDefaultImports": true
```
Comment (mschrage) [inline tsconfig.json:9]: What do these do?

**Review by mschrage — APPROVED** [review]
Looks good to me! Ready to merge :)

File: tsconfig.json (hunk 1 above)
Comment (QuiiBz) [reply tsconfig.json:9]: [noImplicitAny](https://www.typescriptlang.org/tsconfig#noImplicitAny) prevents from not typing variables, and [allowSyntheticDefaultImports](https://www.typescriptlang.org/tsconfig#allowSyntheticDefaultImports) allows to remove the `* as` part from `import` statements if the module doesn't export by default something.

**Review by QuiiBz — APPROVED** [review]
LGTM! Just need to resolve the merge conflict

**Pull request merged**